client.go

```diff
@@ -364,6 +369,8 @@ func (c *Client) runRepliesConsumer(inChan *amqp.Channel) error {
 
 func (c *Client) ensureRunning() {
 	if !c.publisherRunning {
+		c.stopChan = make(chan struct{})
```

Comment:
Not sure I follow, we're not running the publisher more than once even though it's ran in a separate go routine. I cannot see how this would somehow overlap? I guess you're looking for false positives regarding `publisherRunning` or closed channels from other go routines?

Reply:
I’m thinking that if you call Send() from multiple goroutines we might start the connections etc multiple times. Perhaps a read-write mutex on ensureRunning() is better?